src/app.service.ts

```
@@ -1,16 +1,35 @@
 import { Injectable } from '@nestjs/common';
 import { parseGameStateFromFile } from './common/parseConfigFile';
 import { join } from 'path';
-import { GameState } from './common/gameState';
+import {
+  GameState,
+  updateGameState,
+  initNewGameState,
+  DEFAULT_BOARD_SIZE,
+} from './common/gameState';
+import { Coordinates } from './common/utils';
 
 const configPathFromDistDir = '../gameStateFile.json';
 
 @Injectable()
 export class AppService {
-  getBoardStateFromFile(): { gameState: GameState } {
+  getBoardStateFromFile(): GameState {
     const gameState = parseGameStateFromFile(
       join(__dirname, configPathFromDistDir),
     );
-    return { gameState };
+    return gameState;
+  }
+
+  updateGameState(gameState: GameState, coordinates: Coordinates): GameState {
+    try {
+      const updatedGameState = updateGameState(gameState, coordinates);
+      return updatedGameState;
+    } catch {
+      return gameState;
```

Isn't it weird to return the old gamestate if you had a problem? Shouldn't we return an error?